src/routes/sign/Sign.svelte

```diff
@@ -1,7 +1,56 @@
 <script lang="ts">
+    import { sendMessage } from '../../helpers/sendMessageHelpers';
     import Transaction from '../../lib/transaction/Transaction.svelte';
+    import { getParamsFromUrl } from './signHelpers';
+    import { xdr, description, isXdrNull } from './signStore';
+    import type IxdrInvalid from '../../lib/errors/IxdrInvalid';
+    import type { IReadyEvent } from 'src/helpers/eventInterfaces/IReadyEvent';
+
+    let readyEvent: IReadyEvent = {
+        type: 'ready',
+        message: 'Simple Signer is ready to operate',
+    };
+
+    function messageHandler(e: MessageEvent) {
+        if ('xdr' in e.data && 'description' in e.data) {
+            $xdr = e.data.xdr;
+            $description = e.data.description;
+        } else if ('xdr' in e.data) {
+            $xdr = e.data.xdr;
+        }
+    }
+
+    try {
+        const parent = window.opener;
+        const queryString = window.location.search;
+        const urlParams = getParamsFromUrl(queryString);
+
+        if (parent) {
+            sendMessage(readyEvent);
+            window.addEventListener('message', messageHandler);
```

Comment:
this function is registering itself as a message handler? Who is calling this function initially?

Reply:
Sorry, I don't understand, the function is the handler of the message received by the event listener. Should I change it to something like getParamsFromEvent?